plugins/inputs/hugepages/hugepages.go

```diff
@@ -0,0 +1,291 @@
+//go:build linux
+// +build linux
+
+package hugepages
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// path to root huge page control directory
+	rootHugepagePath = "/sys/kernel/mm/hugepages"
+	// path where per NUMA node statistics are kept
+	numaNodePath = "/sys/devices/system/node"
+	// path to the meminfo file
+	meminfoPath = "/proc/meminfo"
+
+	rootHugepages    = "root"
+	perNodeHugepages = "per_node"
+	meminfoHugepages = "meminfo"
+)
+
+var (
+	newlineByte     = []byte("\n")
+	colonByte       = []byte(":")
+	kbPrecisionByte = []byte("kB")
+
+	hugepagesMetricsRoot = []string{
+		"free_hugepages",
+		"nr_hugepages",
+		"nr_hugepages_mempolicy",
+		"nr_overcommit_hugepages",
+		"resv_hugepages",
+		"surplus_hugepages",
+	}
+
+	hugepagesMetricsPerNUMANode = []string{
+		"free_hugepages",
+		"nr_hugepages",
+		"surplus_hugepages",
+	}
+
+	hugepagesMetricsFromMeminfo = []string{
+		"HugePages_Total",
+		"HugePages_Free",
+		"HugePages_Rsvd",
+		"HugePages_Surp",
+		"Hugepagesize",
+		"Hugetlb",
+		"AnonHugePages",
+		"ShmemHugePages",
+		"FileHugePages",
+	}
+)
+
+var hugepagesSampleConfig = `
+  ## Supported huge page types:
+  ##   - "root" - based on root huge page control directory: /sys/kernel/mm/hugepages
+  ##   - "per_node" - based on per NUMA node directories: /sys/devices/system/node/node[0-9]*/hugepages
+  ##   - "meminfo" - based on /proc/meminfo file
+  # hugepages_types = ["root", "per_node"]
```

Comment:
I think we shouldn't prefix the options with `hugepages_` as they are already below the `[inputs.hugepages]` section.
```suggestion
  # types = ["root", "per_node"]
```

Reply:
Sure.